test/gtest/hinge_embedding_loss.hpp

```diff
@@ -0,0 +1,437 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "../driver/tensor_driver.hpp"
+#include "cpu_hinge_embedding_loss.hpp"
+#include "get_handle.hpp"
+#include "miopen/allocator.hpp"
+#include "random.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include <cstddef>
+#include <cstdlib>
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/hinge_embedding_loss.hpp>
+
+struct HingeEmbeddingLossTestCase
+{
+    size_t N;
+    size_t C;
+    size_t D;
+    size_t H;
+    size_t W;
+    float margin;
+    float divisor;
+    std::string reduction;
+    friend std::ostream& operator<<(std::ostream& os, const HingeEmbeddingLossTestCase& tc)
+    {
+        return os << " N:" << tc.N << " C:" << tc.C << " D:" << tc.D << " H:" << tc.H
+                  << " W:" << tc.W << " margin: " << tc.margin;
```

Comment:
I think the divisor, reduction are missed.

Reply:
I have added these fields.

Reply:
@seungmanhan May I ask a question, I saw this function on many gtest file but didn't be used. Is this necessary to have this function, can I delete it?

Reply:
This part is believed to have been used in situations where test case must be output, such as GTEST_SKIP() << config. It was created to immediately check the test cases currently in progress, and is not currently being used. If you don't need it now, I don't think there will be any problem deleting this part.